pkg/pubsub/pubsub_client.go

```diff
@@ -0,0 +1,83 @@
+package pubsub
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/rabbitmq"
+)
+
+func init() {
+	flag.Var(pubSubs, "pub-sub-tool", "Preferred pub-sub tool, e.g dapr, rabbitmq. This flag can be declared more than once. Omitting will default to not using any pub sub tool.")
+}
+
+var pubSubs = newPubSubSet(map[string]PubSub{
+	dapr.Name:     &dapr.Dapr{},
+	rabbitmq.Name: &rabbitmq.RabbitMQ{},
+},
+)
+
+// PubSub is the interface that wraps pubsub methods.
+type PubSub interface {
+	Publish(data interface{}) error
```

Comment:
> @nilekhc - if we should pass context in Publish or not?
> @maxsmythe - Interesting idea. How do the pubsub client libraries use context? 
Generally, I'd expect `context` to have ~the same lifecycle as an outgoing request, so it'd be more natural to use `context` in something like `Publish()` (though batching may complicate that notion)